### PR TITLE
Update: Clarify some contribution guidelines

### PR DESCRIPTION
**Pull request by nielsmh**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,16 +35,16 @@ Good bug reports are extremely helpful, so thanks!
...
 
-2. **Use the GitHub issue search** --- check if the issue has already been
+2. **Use the GitHub issue search** — check if the issue has already been
 reported.
 
-3. **Check if the issue has been fixed** --- try to reproduce it using the latest `nightly` build of OpenTTD, available from https://www.openttd.org
+3. **Check if the issue has been fixed** — try to reproduce it using the latest `nightly` build of OpenTTD, available from https://www.openttd.org
 
-4. **Isolate the problem** --- ideally create reproduceable steps with an attached savegame and screenshots. Try to use few or no NewGRFs, AIs etc if possible.
+4. **Isolate the problem** — ideally create reproduceable steps with an attached savegame and screenshots. Try to use few or no NewGRFs, AIs etc if possible.
```
Comment (andythenorth) [inline CONTRIBUTING.md]: While you're here, 'reproducible' not 'reproduceable'? :)

**Comment by James103** [comment]
In the OpenTTD CI, the Windows and macOS checks are failing in "Install OpenGFX" because `https://cdn.openttd.org/opengfx/0.5.2/opengfx-0.5.2-all.zip` is returning a 404 error. The URL should be changed to `https://cdn.openttd.org/opengfx-releases/0.5.2/opengfx-0.5.2-all.zip` to prevent further errors.

**Comment by nielsmh** [comment]
Yes but that's a fix for #8105 not this PR.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -33,16 +35,16 @@ Good bug reports are extremely helpful, so thanks!
...
 Guidelines for bug reports:
 
-0. Please don't report issues with games where you changed NewGRFs.
+0. Please don't report issues with games where you changed NewGRFs mid-game. (This can be verified with the `gamelog` console command in-game.)
 
-1. Please don't report issues with modified versions of OpenTTD (patchpacks and similar).
+1. Please don't report issues with modified versions of OpenTTD (patchpacks, unofficial ports, and similar).
 
-2. **Use the GitHub issue search** --- check if the issue has already been
+2. **Use the GitHub issue search** — check if the issue has already been
```
Comment (LordAro) [inline CONTRIBUTING.md:42]: Why this change, ooi?

Comment (nielsmh) [reply CONTRIBUTING.md:42]: When viewed in proportional fonts, a proper em-dash looks better than three hyphens. So entirely cosmetic.

**Comment by andythenorth** [comment]
I'm assuming the checks are stuck.  Because this diff is clearly docs-only, I'm approving it.

**Review by andythenorth — APPROVED** [review]

**Review by LordAro — APPROVED** [review]
